qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/ClientCancelPendingCommandTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.it.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.gateway.metrics.LongPollingMetrics;
+import io.camunda.zeebe.it.clustering.ClusteringRuleExtension;
+import java.time.Duration;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+final class ClientCancelPendingCommandTest {
+  @RegisterExtension
+  final ClusteringRuleExtension cluster = new ClusteringRuleExtension(1, 1, 1, cfg -> {});
+
+  @Test
+  void shouldCancelCommandOnFutureCancellation() {
+    // given
+    final var client = cluster.getClient();
+    final var future =
+        client
+            .newActivateJobsCommand()
+            .jobType("type")
+            .maxJobsToActivate(10)
+            .requestTimeout(Duration.ofHours(1))
+            .send();
+    final var metrics = new LongPollingMetrics();
+    Awaitility.await("until we have one polling client")
+        .untilAsserted(() -> assertThat(metrics.getBlockedRequestsCount("type")).isOne());
+
+    // when - create some jobs after cancellation; the notification will trigger long polling to
+    // remove cancelled requests. unfortunately we can't tell when cancellation is finished
+    future.cancel(true);
+
+    // then
+    Awaitility.await("until no long polling clients are waiting")
+        .untilAsserted(() -> assertThat(metrics.getBlockedRequestsCount("type")).isZero());
```

Comment:
:thinking: I've been trying to see if there's another way to assert this.

Our goal would be to have access to the `jobTypeState` Map inside the `LongPollingActivateJobsHandler`. Unfortunately, the property and all methods that use it are `private`. Furthermore, we can't really access the `LongPollingActivateJobsHandler` instance from the `Gateway` since it's always handled as a local variable inside private methods.

Maybe we can create a "RecordingLongPollingActivateJobsHandler" that exposes the properties we need, and hook it up in the gateway. We would need to modify runtime code to achieve this. 

So finally, I'm wondering if the current solution is enough, especially since our goal is to phase out job polling in the future.

Alternatively, if we're able to `Mockito#spy` on a `LongPollingActivateJobsHandler` instance, we might be able to achieve something nicer in the test as well. But still, this approach would require us changing runtime code.

Reply:
Regarding your suggestions, all of them have the same downsides as the metrics based approach =/ The gateway is unfortunately too black-boxy for us to do any kind of dependency injection at the QA/acceptance level.

Reply:
Let's revisit this once we have job streaming. I think we'll have better QA tests then, and it might be enough to have those at the QA level, and everything else at the unit level.